### ISSUE TITLE
Drop in either `atexit` or `std::signal` causes core dumped.

### ISSUE THREAD

**Issue by YuanYuYuan** [eclipse-zenoh/zenoh-cpp#198]
### Describe the bug

skip

### To reproduce

skip

### System info

skip

**Comment by yellowhatter** [comment]
There are some (a lot of) system APIs that should not be called in signal handlers, so it is really a bad idea to drop big objects in signal handlers. Moreover, signal handler can be executed concurrently, so thread safety also comes into place. We need to handle signals in a safe way, I can take this task

**Comment by YuanYuYuan** [comment]
> There are some (a lot of) system APIs that should not be called in signal handlers, so it is really a bad idea to drop big objects in signal handlers. Moreover, signal handler can be executed concurrently, so thread safety also comes into place. We need to handle signals in a safe way, I can take this task

Yes. I agree with your point. Dropping objects within signal handlers isn't a good pattern. Rust has put much effort into addressing this safely within the normal scope.

**Comment by DenisBiryukov91** [comment]
The issue is not related to zenoh-c/cpp but to zenoh rust implementation relying on tis variables.
